store/index/gc.go

```diff
@@ -72,39 +112,44 @@ func (i *Index) garbageCollector(gcInterval time.Duration) {
 	}
 }
 
-// gc searches for and removes stale index files. Returns the number of unused
-// index files that were removed.
-func (i *Index) gc(ctx context.Context) (int, error) {
-	header, err := readHeader(i.headerPath)
+// cycle searches for and removes stale index files. Returns the number of
+// unused index files that were removed.
+func (gc *indexGC) Cycle(ctx context.Context) (int, error) {
+	gc.cycleLock.Lock()
+	defer gc.cycleLock.Unlock()
+
+	gc.cycleCount++
+
+	header, err := readHeader(gc.index.headerPath)
 	if err != nil {
 		return 0, err
 	}
 	fileNum := header.FirstFile
 
 	// Before scanning the index files, check if the first index file is still
 	// in use by the bucket index last seen using it.
-	if i.gcCheckpoint {
-		inUse, err := i.bucketInFile(i.gcBucketIndex, fileNum)
+	if gc.checkpoint {
+		inUse, err := gc.bucketInFile(gc.bucketIndex, fileNum)
 		if err != nil {
-			i.gcCheckpoint = false
+			gc.checkpoint = false
 			return 0, err
 		}
 		if inUse {
 			// First index file still used to store info.
 			return 0, nil
 		}
 		// Checkpoint bucket checked.
-		i.gcCheckpoint = false
+		gc.checkpoint = false
 	}
 
 	var count int
 	for {
```

Comment:
Instead of running through all the primary files, it may be better to only collect from a limited number (1?) or only collect for a limited time. Consider exiting the loop after some time limit (how long?).